code/game/machinery/doors/airlock.dm

```diff
@@ -799,10 +758,10 @@
 
 
 /obj/machinery/door/airlock/attackby(obj/item/C, mob/user, params)
-	if(!issilicon(user) && !IsAdminGhost(user))
-		if(isElectrified())
-			if(shock(user, 75))
-				return
+	if(prob(75))
+		if(SEND_SIGNAL(src, COMSIG_AIRLOCK_ATTACKBY_CONDUCTIVE, user) & COMSIG_ELECTRIFIED_SHOCK)
```

Comment:
1. The component should be deciding when it shocks
2. The signal should be sent every time if you're going to have this
3. Call parent and use the base attackby signal instead